src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -1980,46 +1980,34 @@ private ConsumeTokenResult ConsumeNextTokenFromLastNonCommentToken()
             }
             else if (_tokenType == JsonTokenType.StartObject)
             {
-                if (first == JsonConstants.CloseBrace)
```

Comment:
Note: per offline discussion this was unreachable to begin with (so not part of the issue for other reviewers)